src/main/java/org/springframework/retry/annotation/Backoff.java

```diff
@@ -130,7 +131,10 @@
 	 * this to true to have the backoff delays randomized, so that the maximum delay is
 	 * multiplier times the previous delay and the distribution is uniform between the two
 	 * values. Use {@code #{...}} for one-time evaluation during initialization, omit the
-	 * delimiters for evaluation at runtime.
+	 * delimiters for evaluation at runtime. This expression is always evaluated at
+	 * configuration phase. If the expression returns true then
+	 * {@link org.springframework.retry.backoff.ExponentialRandomBackOffPolicy} is used
+	 * else {@link org.springframework.retry.backoff.ExponentialBackOffPolicy} is used.
```

Comment:
This is still not fixed. The `random` is always evaluated on configuration phase. We have to not its result to determine the target `BackOffPolicy`.

Reply:
If you look at the following code: the policy is being determined by the expression's evaluation. So the above statement makes sense right?
`if (isRandom()) {
				policy = new ExponentialRandomBackOffPolicy();
			}
			
private boolean isRandom() {
		return (this.randomSupplier != null && Boolean.TRUE.equals(this.randomSupplier.get()))
				|| Boolean.TRUE.equals(this.random);
	}`

Reply:
It is, but read this whole Javadoc again.
I worry about the sentence:

```
Use {@code #{...}} for one-time evaluation during initialization, omit the delimiters for evaluation at runtime
```
And that is exactly what I'm asking you to fix.
The `random` expression is never evaluated at runtime.
Instead it is evaluated at configuration phase unconditionally.

Thanks